src/main/java/uk/gov/hmcts/reform/authorisation/generators/ServiceAuthTokenGenerator.java

```diff
@@ -0,0 +1,31 @@
+package uk.gov.hmcts.reform.authorisation.generators;
+
+import com.warrenstrange.googleauth.GoogleAuthenticator;
+import uk.gov.hmcts.reform.authorisation.ServiceAuthorisationApi;
+
+import static java.lang.String.format;
+
+public class ServiceAuthTokenGenerator implements AuthTokenGenerator {
+
+    private final String secret;
+    private final String microService;
+    private final ServiceAuthorisationApi serviceAuthorisationApi;
+    private final GoogleAuthenticator googleAuthenticator;
+
+    public ServiceAuthTokenGenerator(
+            final String secret,
+            final String microService,
+            final ServiceAuthorisationApi serviceAuthorisationApi
+    ) {
+        this.secret = secret;
+        this.microService = microService;
+        this.serviceAuthorisationApi = serviceAuthorisationApi;
+        this.googleAuthenticator = new GoogleAuthenticator();
+    }
+
+    @Override
+    public String generate() {
+        final String oneTimePassword = format("%06d", googleAuthenticator.getTotpPassword(secret));
+        return "bearer " + serviceAuthorisationApi.serviceToken(this.microService, oneTimePassword);
```

Comment:
Should be capital 'B' I think.
If it's actually needed, IMHO it should be just a token. @timja-kainos ?

Reply:
header parsing should be done case insensitively, http2 is lower case I believe. 
imo it is correct as is although convention is usually for it to be upper case.

but...
Imo generate shouldn't add the `Bearer <space>`,
Bearer isn't part of the token, it is part of the auth header format

If this was `ServiceAuthHeaderGenerator` then it would be appropriate 😉

Reply:
Cool, let's return just the token as the class name suggest and add `"Bearer "` prefix when constructing a header to s2s api.